lib/datasource.js

```diff
@@ -781,6 +781,18 @@ DataSource.prototype.supportTypes = function(types) {
 };
 
 /**
+ * Generate random id.  Each data source model must override this method.
+ * @returns {<model dependent>} Data type varies from model to model,
+ */
+
+DataSource.prototype.generateUniqueId = function() {
```

Comment:
The `generateUinqueId` should take the `model` so that it can generate the id based on the ID definition of the model.